core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/fuse/PlanNodeFuser.java

```diff
@@ -0,0 +1,530 @@
+     * <pre>
+     * SELECT i_item_sk AS sk, i_brand AS brand
+     * FROM item
+     *
+     * SELECT i_brand AS brand2, i_size AS size
+     * FROM item
+     * </pre>
+     * results in a fused result (P, M, TRUE, TRUE) where:
+     * <pre>
+     * P: SELECT
+     *   i_item_sk AS sk,
+     *   i_brand AS brand,
+     *   i_size AS size
+     * FROM item
+     * M: brand2 → brand
+     * </pre>
+     * </p>
+     */
+    private Optional<FusedPlanNode> fuseTableScan(TableScanNode left, TableScanNode right)
+    {
+        if (!left.getTable().equals(right.getTable())) {
+            return Optional.empty();
+        }
+
+        if (!left.getEnforcedConstraint().equals(right.getEnforcedConstraint()) ||
+                !left.getStatistics().equals(right.getStatistics()) ||
+                !left.isUpdateTarget() == right.isUpdateTarget() ||
+                !left.getUseConnectorNodePartitioning().equals(right.getUseConnectorNodePartitioning())) {
+            return Optional.empty();
+        }
+
+        Set<Symbol> outputs = new HashSet<>(left.getOutputSymbols());
+        Map<Symbol, ColumnHandle> assignments = new HashMap<>(left.getAssignments());
+        FusedSymbolMapping.Builder symbolMappingBuilder = FusedSymbolMapping.builder();
+        Set<Symbol> rightOutputSymbols = ImmutableSet.copyOf(right.getOutputSymbols());
+
+        // add unique assignments and outputs from the right side or map to the left side symbols
+        right.getAssignments().forEach(((symbol, columnHandle) ->
+                left.getAssignments().entrySet().stream().filter(entry -> entry.getValue().equals(columnHandle)).findFirst()
+                        .ifPresentOrElse(
+                                entry -> {
+                                    // left assignment found, add the mapping and output symbol if necessary
+                                    symbolMappingBuilder.add(symbol, entry.getKey());
+                                    if (rightOutputSymbols.contains(symbol)) {
+                                        outputs.add(entry.getKey());
+                                    }
+                                },
+                                () -> {
+                                    // left assignment not found, add a new assignment and output symbol if necessary
+                                    assignments.put(symbol, columnHandle);
+                                    if (rightOutputSymbols.contains(symbol)) {
+                                        outputs.add(symbol);
+                                    }
+                                })));
+        return Optional.of(new FusedPlanNode(
+                new TableScanNode(
+                        context.getIdAllocator().getNextId(),
+                        left.getTable(),
+                        ImmutableList.copyOf(outputs),
+                        assignments,
+                        left.getEnforcedConstraint(),
+                        left.getStatistics(),
```

Comment:
That link didn't work for me.
From what I can tell, since we are doing this before predicate pushdown, `outputRowCount` should be the same for both scans, so it should be the same for the fused scan as well. When adding symbols to `outputs` above, we can also lookup the symbol stats for that from whichever TS it is part of and use that to build symbol stats of the fused scan.

Reply:
> That link didn't work for me.

this should work. https://github.com/trinodb/trino/pull/14271#discussion_r1009257047

Instead of merging stats hereI added preconditions that stats are empty.